www/js/survey/launch.js

```diff
@@ -107,19 +113,23 @@ angular.module('emission.survey.launch', ['emission.services',
           .then(function(event) {
             console.log("successfully opened page with result "+JSON.stringify(event));
             // success
-            replace_uuid(uuidElementId)
-            .catch(function(error) {
-              $ionicPopup.alert({"template": "Relaunching survey - while replacing uuid, got error "+ JSON.stringify(error)})
-              .then(function() {
-                surveylaunch.startSurvey(url, uuidElementId);
-              });
-            });
+            //replace_uuid(uuidElementId)
           })
           .catch(function(event) {
             // error
           });
       $rootScope.$on('$cordovaInAppBrowser:loadstart', function(e, event) {
         console.log("started loading, event = "+JSON.stringify(event));
+
+
+      });
+      $rootScope.$on('$cordovaInAppBrowser:loadstop', function(e, event) {
+        console.log("stopped loading, event = "+JSON.stringify(event));
+        if (String(event.url).includes('formResponse')) {
+            $cordovaInAppBrowser.close();
+        }
+        replace_uuid(uuidElementId);
```

Comment:
so you decided to replace the uuid on load stop instead of load start?
Why? Did you test to confirm that it worked?

Reply:
Yes, I just decided to move all of the logic to loadstop.
The auto-close is on load stop because when it's in load start the survey actually doesn't get submitted. 

So I just decided to put replace_uuid to load stop also so all modifications to the inAppBrowser function happen after the loading stops (and it is working).